sql/driver/driver_test.go

```diff
@@ -385,28 +386,38 @@ func TestConnectionSettings(t *testing.T) {
 	}
 }
 
-func TestInsecure(t *testing.T) {
+func TestProtocols(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	// Start test server in insecure mode.
-	s := &server.TestServer{}
-	s.Ctx = server.NewTestContext()
-	s.Ctx.Insecure = true
-	if err := s.Start(); err != nil {
-		t.Fatalf("Could not start server: %v", err)
-	}
-	defer s.Stop()
-
-	// We can't attempt a connection through HTTPS since the client just retries forever.
-	// DB connection using plain HTTP.
-	db, err := sql.Open("cockroach", "http://root@"+s.ServingAddr())
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer func() {
-		_ = db.Close()
-	}()
-	if _, err := db.Exec(`SELECT 1`); err != nil {
-		t.Fatal(err)
+
+	// Test that all of the network protocols work.
+	for _, scheme := range []string{"http", "https", "rpc", "rpcs"} {
```

Comment:
FWIW we already have constants for these in base/context.go; I think it'd be better to export them there so this doesn't rot.